packages/fetch-http-handler/src/fetch-http-handler.ts

```diff
@@ -185,7 +185,11 @@ export class FetchHttpHandler implements HttpHandler<FetchHttpHandlerConfig> {
         })
       );
     }
-    return Promise.race(raceOfPromises).finally(removeSignalEventListener);
+    return Promise.race(raceOfPromises).finally(() => {
+      if (typeof removeSignalEventListener === "function") {
```

Comment:
initialize `removeSignalEventListener = () => {}` instead?

could you add a comment that this is only needed in react-native?

Reply:
> initialize `removeSignalEventListener = () => {}` instead?

Verified that initializing `removeSignalEventListener` as an empty function works
https://github.com/aws/aws-sdk-js-v3/issues/6269#issuecomment-2253018138

Reply:
Posted a new fix in https://github.com/smithy-lang/smithy-typescript/pull/1351